Day-02/kotlin/StckOverflw/Part1.kt

```diff
@@ -0,0 +1,17 @@
+fun main() {
+    val input = {}.javaClass.getResource("/input.txt")?.readText() ?: error("No input.txt in resources")
+
+    var horizontal = 0
+    var depth = 0
+
+    input.split("(\\r\\n|\\r|\\n)".toRegex()).forEach {
+        val (instructionName, value) = it.split(' ')
+        if (instructionName.equals("forward", true)) {
+            horizontal += value.toInt()
+        } else {
+            depth += if (instructionName.equals("down", true)) value.toInt() else -value.toInt()
+        }
+    }
+
```

Comment:
This would be a nice use case of the `[fold](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/fold.html)` method like I used here

https://github.com/DRSchlaubi/aoc-kotlin-2021/blob/f3f6746086ad522efa5690cc647fc75a270cfa48/src/main/kotlin/day_two/Shared.kt#L31-L43